pkg/autoscaler/autoscaler.go

```diff
@@ -597,10 +616,21 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 			Consistently(func() (bool, error) {
 				nodes, err := framework.GetReadyAndSchedulableNodes(client)
 				return len(nodes) == caMaxNodesTotal, err
-			}, framework.WaitShort, pollingInterval).Should(BeTrue())
+			}, framework.WaitShort, pollingInterval).Should(
+				BeTrue(),
+				"Expected to reach %d nodes, observed %d nodes",
+				caMaxNodesTotal,
+				func() int {
+					nodes, err := framework.GetReadyAndSchedulableNodes(client)
+					if err != nil {
+						return 0
+					}
+					return len(nodes)
+				})
```

Comment:
Will gomega execute this function when there's a failure? I think we can simplify these assertions
```
			Consistently(func() (int, error) {
				nodes, err := framework.GetReadyAndSchedulableNodes(client)
				return len(nodes), err
			}, framework.WaitShort, pollingInterval).Should(
				Equal(caMaxNodesTotal),
```
If we do this, we don't need to put the additional context into the error messages and can rely on gomega printing out the failed 2 = 4 or whatever it is

Reply:
> Will gomega execute this function when there's a failure?

that was my understanding from looking at the gomega docs

i'm fine to put more of this into the context, it wasn't clear to me how returning values from the inner, tested, function would propagate to the log lines.